src/RegisterDevicePage.tsx

```diff
@@ -121,44 +152,100 @@ const IntrospectionControlRow = ({
+        minor: interfaceDetail.version_minor,
+      }));
+      setSelectedInterfaceData(fetchedInterfaces);
+      setGreaterRealmManagementVersion(true);
+      return fetchedInterfaces;
+    } else {
+      setGreaterRealmManagementVersion(false);
+    }
+  }, [astarte.client]);
+
+  useEffect(() => {
+    const loadOptions = async () => {
+      const interfaces = await fetchInterfacesInfo();
+      if (interfaces) {
+        const interfaceOptions = interfaces.map((iface) => ({
+          value: iface.name,
+          label: iface.name,
+        }));
+        setOptions(interfaceOptions);
+      }
+    };
+    loadOptions();
+  }, [fetchInterfacesInfo]);
+
   return (
     <tr>
       <td>
-        <Form.Control
-          type="text"
-          placeholder="Interface name"
-          value={interfaceDescriptor.name}
-          onChange={handleNameChange}
-        />
-      </td>
-      <td>
-        <Form.Control
-          type="number"
-          min="0"
-          value={interfaceDescriptor.major}
-          onChange={handleMajorChange}
-        />
-      </td>
-      <td>
-        <Form.Control
-          type="number"
-          min="0"
-          value={interfaceDescriptor.minor}
-          onChange={handleMinorChange}
-        />
-      </td>
-      <td>
-        <Button
-          variant="secondary"
-          disabled={interfaceDescriptor.name === ''}
-          onClick={() => {
-            onAddInterface(interfaceDescriptor);
-            setInterfaceDescriptor(initialState);
-          }}
-        >
-          Add
-        </Button>
+        {greaterRealmManagementVersion ? (
+          <Form.Select
+            value={selectedOption}
+            onChange={(e) =>
+              handleNameChange((e.target.selectedIndex - 1).toString(), ChangeSource.SelectOptions)
+            }
+            className="form-control"
+          >
+            <option value="" disabled>
+              Interface name
+            </option>
+            {options.map((option, index) => (
+              <option key={index} value={index.toString()}>
+                {option.label}
+              </option>
+            ))}
+          </Form.Select>
+        ) : (
+          <Form.Control
+            type="text"
+            placeholder="Interface name"
+            value={interfaceDescriptor.name}
+            onChange={(e) => handleNameChange(e.target.value, ChangeSource.Input)}
+          />
+        )}
       </td>
+      {(hasSelectedInterface || !greaterRealmManagementVersion) && (
+        <>
+          <td>
+            <Form.Control
+              type="number"
+              min="0"
+              value={interfaceDescriptor.major}
+              onChange={handleMajorChange}
+            />
+          </td>
+          <td>
+            <Form.Control
+              type="number"
+              min="0"
+              value={interfaceDescriptor.minor}
+              onChange={handleMinorChange}
+            />
+          </td>
```

Comment:
Those work great when the user has to manually select the interface version (prior Astarte 1.3).
When the user selects one of the interfaces through the dropdown (after Astarte 1.3) the user also selects a specific major, so it should not be edited afterwards.